integrated_channels/blackboard/migrations/0005_blackboardlearnerassessmentdatatransmissionaudit.py

```diff
@@ -15,7 +15,7 @@ class Migration(migrations.Migration):
             fields=[
                 ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                 ('blackboard_user_email', models.CharField(max_length=255)),
-                ('enterprise_course_enrollment_id', models.PositiveIntegerField(db_index=True)),
+                ('enterprise_course_enrollment_id', models.IntegerField(db_index=True)),
```

Comment:
It requires for github migration check otherwise it will break with long check name error.

Reply:
Changing existing migrations is dangerous, and I'm not sure it is wanted in this case. If you were to change this, some deployments would have an IntegerField and some would have a PositiveIntegerField when you get to the later AlterField. Does the AlterField really do the right thing in both cases (e.g. remove the constraint for a PositiveIntegerField, and act as a no-op with no error for the IntegerField? I would think this wouldn't create the SQL to remove the constraint if it is going from an IntegerField to IntegerField.

Instead, if you add the AlterField, does the problem go away as soon as you squash migrations? Or, not until you get to phase 2 of squashing, when you can finally delete the old migrations?

If squashing (phase 1) is enough (and that's an open question for me), could we instead do the following:
1. Add the AlterField.
2. Squash migrations.
3. Celebrate no error?!
4. Delete old migrations after the next named release?

Note: If the problem isn't solved until the migration is actually deleted, this plan won't work unless the mysql8 upgrade can wait until after the next named release, because we would need to wait for the next named release before deleting the old migrations.

Reply:
@robrap I did these steps now 

1. Add the AlterField 
2. Generate new migration for alter field
2. Squash migration.
3. Check the hash against mysql8-check migrations in [platform](https://github.com/openedx/edx-platform/runs/4951340194?check_suite_focus=true#step:11:168) ( failed )

Squash migration includes this https://github.com/openedx/edx-enterprise/pull/1446/files#diff-82ed9615cb2cf73f24651c57f6b548ae58b7a99ca64900d4054b38e505f5381aR57 and when it run on mysql8 it crashes.

Reply:
In case of squash this issue will appear because it will pick the `PositiveIntegerField` from initial migration how to handle this ?

1. Does the AlterField really do the right thing in both cases (e.g. remove the constraint for a PositiveIntegerField) **Yes it will solve the issue **
2. act as a no-op with no error for the IntegerField to IntegerField ( **no idea need to do discovery** )

Reply:
@awais786: The squash migration is auto-generated, but you can manually update it before merging. You would just remove the `AlterField` and copy the change into the `CreateModel`. This is basically what you were wishing to do, but I think doing it inside the squash migration is the place to do it. Thus, any deployment that has some, but not all, of the squashed migrations would just run the rest of the non-squashed migrations, including the AlterField. However, any new deployment (or deployment that hasn't gotten to the replaced migrations) would instead run the squashed migration that would never refer to `PositiveIntegerField`. Hopefully, it is the latter case that would run for `mysql8-check` and would pass.